src/cpp/include/openvino/genai/streamer_base.hpp

```diff
@@ -4,16 +4,28 @@
 #pragma once
 
 #include "openvino/genai/tokenizer.hpp"
+#include <variant>
 
 namespace ov {
 namespace genai {
 
+enum class StreamerRunningStatus {
+    UNDEF = 0, // Streaming is not run
+    RUNNING = 1, // Continue to run of inference
+    STOP = 2, // Stop generation, keep history as is, KV cache includes last request and generated tokens
+    CANCEL = 3 // Stop generate, drop last prompt and all generated tokens from history, KV cache include history but last step
+};
+
+using CallbackTypeVariant = std::variant<bool, StreamerRunningStatus>;
```

Comment:
```suggestion
using CallbackTypeVariant = std::variant<void, bool, StreamerRunningStatus>;
```
to support callback which just "prints"